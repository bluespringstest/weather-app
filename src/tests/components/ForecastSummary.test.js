import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
    // const date = new Date('Tuesday 2 April');
    const validProps = {
        date: 1525046400000,
        description: "Stub Description",
        icon: "800",
        temperature: {
            min: 12,
            max: 22,
        },
        onSelect: jest.fn(),
    };
    
    it("renders the objects correctly in the virtual DOM", () => {
        const { asFragment } = render(
            <ForecastSummary 
            date={validProps.date}
            description={validProps.description}
            icon={validProps.icon}
            temperature={validProps.temperature}
            onSelect={validProps.onSelect}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it("renders the correct values for the props", () => {
        const { getByText, getByTestId } = render(
            <ForecastSummary 
            date={validProps.date}
            description={validProps.description}
            icon={validProps.icon}
            temperature={validProps.temperature}
            onSelect={validProps.onSelect}
            />
        );
        expect(getByText('Mon 1st Apr')).toHaveClass("forecast-summary__date");
        expect(getByText("Stub Description")).toHaveClass("forecast-summary__description");
        expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
        expect(getByText("22Â°C")).toHaveClass("forecast-summary__temperature");
    });
});