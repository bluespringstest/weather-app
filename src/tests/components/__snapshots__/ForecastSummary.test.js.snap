// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders the objects correctly in the virtual DOM 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      Mon 1st Apr
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      800
    </div>
    <div
      class="forecast-summary__temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub Description
    </div>
    <button
      type="button"
    >
      More details
    </button>
  </div>
</DocumentFragment>
`;
